public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx

```diff
@@ -30,42 +30,58 @@ export const MatcherFilter = ({ className, onFilterChange, defaultQueryString }:
   useEffect(() => onSearchInputChanged.cancel(), [onSearchInputChanged]);
 
   const searchIcon = <Icon name={'search'} />;
+  const inputInvalid =
+    defaultQueryString && defaultQueryString.length > 3 ? parseMatchers(defaultQueryString).length === 0 : false;
```

Comment:
What's the reason for checking the length of the input here? 

The `{` and `}` are optional for the input.

Reply:
> What's the reason for checking the length of the input here?

I was about to ask you as you added the same expression in this [PR](https://github.com/grafana/grafana/pull/61952/files#diff-bd5675bd3c94c3fc3286df05fd93b5cba420dd8a6b23c70db2b5e69800f4c96aR51)

I wanted to be consistent with this logic that is in different filters (Silences also use this expression)

Reply:
> The { and } are optional for the input.

yes

Reply:
Resolving this; it's probably been copied from various other places for a while and might have been there for historical reasons that are no longer relevant.